### PR TITLE
AutoTotem now sets your offhand to Totem

### PR DESCRIPTION
**Pull request by Vonr**
I don't know how to make settings so don't bother asking me to make them thanks.

### CODE REVIEW

**Comment by ghost** [comment]
AutoTotem worked before and its not supposed to spawn in a new totem

**Comment by Vonr** [comment]
make it an option if you can be bothered to, else wait another 2 days and ill learn how to make it myself.

**Comment by hYdos** [comment]
its not that hard look at other modules

**Comment by Vonr** [comment]
uh, weird shit here but apparently the normal autototem now also sets offhand, someone check my code for weird shit please

**Comment by Vonr** [comment]
also theres a useless if condition for checking if the item is null but i cant be bothered to push a commit until i fix the setoffhand glitch

**Comment by Vonr** [comment]
i think the problem is that it doesnt consume the totem

**Comment by Vonr** [comment]
why does that have to automatically go into the pr... well ill find a way to stop that i guess

**Comment by Vonr** [comment]
the autototem bug is not my fault. ill create an issue on it.

**Comment by Vonr** [comment]
Requesting for merge. code is all okay

File: AutoTotem.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#include "AutoTotem.h"
```
Comment (ghost) [inline AutoTotem.cpp]: why is there a new file in the root folder

File: Horion/Module/Modules/AutoTotem.cpp [hunk 2]
```diff
@@ -21,29 +22,51 @@ void AutoTotem::onTick(C_GameMode* gm) {
...
-				C_ItemStack* test = a->getItemStack(i);
-				if (test->item != NULL) {
-					C_Item* yikes = *test->item;
-					if (yikes->itemId == 450) {
-						g_Data.getLocalPlayer()->consumeTotem();
-						g_Data.getLocalPlayer()->setOffhandSlot(test);
-					}
-
-				}
+			if (isSOH == true)
```
Comment (ghost) [inline Horion/Module/Modules/AutoTotem.cpp]: add this check here as well: `i->item == NULL`

**Review by ghost — CHANGES_REQUESTED** [review]

File: AutoTotem.cpp (hunk 1 above)
Comment (Vonr) [reply AutoTotem.cpp]: fixed

File: Horion/Module/Modules/AutoTotem.cpp (hunk 2 above)
Comment (Vonr) [reply Horion/Module/Modules/AutoTotem.cpp]: i havent tested with the SOH autototem, but im pretty sure that that check messes stuff up if there is an item clientsidedly but not serversidedly (visual flickering for example) ill add it regardless

**Comment by ghost** [comment]
sir pls accept the fucking PR

**Comment by SurvirvorHRG** [comment]
Why not

**Comment by Vonr** [comment]
pullitalready

**Comment by Vonr** [comment]
ಠ_ಠ ʘ‿ʘ (╯°□°）╯︵┻━┻

**Comment by Vonr** [comment]
ლ(ﾟДﾟლ) BOI PULL IT

**Comment by richardletshacks** [comment]
@SurvirvorHRG is everything okay now, can we merge this?

**Comment by ghost** [comment]
smh just closed without pull :(

**Comment by ghost** [comment]
Sir, fuck you and please fucking pull it first.

**Comment by Greloom-Yeet** [comment]
SIR, ACCEPT THE FUCKING PULL REQUEST

**Comment by ghost** [comment]
Sir, this piece of software instructions designed to give the user an advantage in a game known widely as Minecraft should be merged into the main, central repository. As such, I would like to request that YOU FUCKING PULL IT.

**Pull request closed without merging**